File: frontend/types/application.ts
import type { Readiness } from './intelligence';

export type ApplicationStatus =
  | 'discovered'
  | 'preparing'
  | 'ready'
  | 'submitted'
  | 'under_review'
  | 'additional_information_required'
  | 'approved'
  | 'rejected'
  | 'completed';

export interface ApplicationEvent {
  oldStatus: string | null;
  newStatus: string;
  source: string;
  note: string | null;
  at: string;
}

export interface Application {
  id: string;
  serviceId: string;
  status: ApplicationStatus;
  notes: string | null;
  submittedAt: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface ApplicationDetail extends Application {
  readiness: Readiness | null;
  timeline: ApplicationEvent[];
}

File: backend/app/schemas/application.py
from pydantic import BaseModel

from app.schemas.intelligence import ReadinessRead


class ApplicationCreate(BaseModel):
    serviceId: str


class ApplicationStatusUpdate(BaseModel):
    status: str
    note: str | None = None


class ApplicationEventRead(BaseModel):
    oldStatus: str | None = None
    newStatus: str
    source: str
    note: str | None = None
    at: str


class ApplicationRead(BaseModel):
    id: str
    serviceId: str
    status: str
    notes: str | None = None
    submittedAt: str | None = None
    createdAt: str
    updatedAt: str


class ApplicationDetailRead(ApplicationRead):
    readiness: ReadinessRead | None = None
    timeline: list[ApplicationEventRead] = []
